libbeat/common/file/file_windows.go

```diff
@@ -43,7 +44,13 @@ func (fs StateOS) IsSame(state StateOS) bool {
 }
 
 func (fs StateOS) String() string {
-	return fmt.Sprintf("%d-%d-%d", fs.IdxHi, fs.IdxLo, fs.Vol)
+	var buf [92]byte
```

Comment:
I wonder how much this improves performance compared to more readable code. I would not expect this to make a big difference in the overall performance.

Reply:
Quite hughe difference if registry file is big. With hughe registry file ID generation with fmt.Sprintf is one of the worst contenders in a CPU profile (at least during startup).

Follow up PR with String builder optimization in #6343